### DISCUSSION TITLE
In Flask-Socketio, how can we save the session of the connected client.

### DISCUSSION THREAD

**Discussion by neerajdlh74** [miguelgrinberg/python-socketio#1044] [Q&A]
Can someone help me how i can save the session details of the client in Flask-socketio like we do in python-socketio framework? And how i can access it afterwards on request from client or in some other method?
And i also observed saved_session key in request.environ object, how this can be useful?

**Comment by miguelgrinberg** [comment]
Flask-SocketIO uses the Flask user session, not the session feature provided by python-socketio.

The documentation has a section that explains how `session` and other Flask context locals work: https://flask-socketio.readthedocs.io/en/latest/implementation_notes.html#access-to-flask-s-context-globals.

There is also a [video](https://www.youtube.com/watch?v=OZ1yQTbtf5E) that I recorded that is relevant.
